workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/container/LayeredDialogSessions.kt

```diff
@@ -113,7 +113,35 @@ public class LayeredDialogSessions private constructor(
 
   private var sessions: List<DialogSession> = emptyList()
 
-  public var allowEvents: Boolean = true
+  /**
+   * Reflects whether [onAttachedToWindow] has been called yet, so that
+   * we can avoid [showing][DialogSession.showNewDialog] `Dialog` windows
+   * before the `Activity` window is ready.
+   */
+  private var attached = false
+
+  /**
+   * Set by an outermost instance only. If we are in a root [update] call before
+   * [attached] is true, holds on to a list set of all [DialogSession]s, to be
+   * [shown][DialogSession.showNewDialog] at attach time instead of immediately.
+   * Necessary to ensure that view state is restored in the correct order --
+   * the `Activity` `contentView` must be attached to its window before that of
+   * any `Dialog`.
+   *
+   * Note that this is a superset of [sessions], also includes sessions built by
+   * any nested [LayeredDialogSessions] instances during recursive [update]
+   * calls.
+   *
+   * https://github.com/square/workflow-kotlin/issues/1001
+   */
+  private var deferredShow: List<DialogSession>? = null
+
+  /**
+   * Clients should ignore UI events targeting their [content][ScreenOverlay.content]
```

Comment:
so this is where the `content` naming comes from? Hmm. How do I map body to content? are they equivalent? 'Body' is the container for 'content'?

Reply:
Oh, derp: and that kdoc is TOTALLY WRONG! Link should be to `BodyAndOverlays.content`!

Reply:
Tell you what, let's decouple this naming conversation from the ordering fix. Maybe even move it upstream.

Reply:
```kotlin
  /**
   * While this is true, clients should ignore UI events targeting the
   * [body][BodyAndOverlaysScreen.body] view beneath the
   * [Dialog][android.app.Dialog] windows they manage. It reflects
   * that a [ModalOverlay] is (or soon will be) showing over the body screen.
   */
  public var allowBodyEvents: Boolean = true
```